lib/elixir/lib/calendar.ex

```diff
@@ -338,6 +338,31 @@ defmodule Calendar do
   @doc since: "1.15.0"
   @callback iso_days_to_end_of_day(iso_days) :: iso_days
 
+  @doc """
+  Shifts date by given duration according to its calendar.
+  """
+  @callback shift_date(year, month, day, Duration.t()) :: {year, month, day}
```

Comment:
Oh, before I forget, let's add `@doc since: "1.17.0"` to all new callbacks and functions.